cmake/TaichiCXXFlags.cmake

```diff
@@ -79,14 +79,13 @@ endif()
     # However, some of these "constexpr" are debug flags and will be manually enabled upon debugging.
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-unneeded-internal-declaration ")
 
-    # CLANG_VERSION_MAJOR is set in check_clang_version
-    if (${CLANG_VERSION_MAJOR} VERSION_GREATER_EQUAL 15)
-      # [Global] FIXME: Newly introduced flag in clang-15
+    check_cxx_compiler_flag("-Wno-unqualified-std-cast-call" CXX_HAS_Wno_unqualified_std_cast_call)
+    if (${CXX_HAS_Wno_unqualified_std_cast_call})
       set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-unqualified-std-cast-call ")
     endif()
 
-    if (${CLANG_VERSION_MAJOR} VERSION_GREATER_EQUAL 13)
```

Comment:
This is the clang compiler version for the runtime, it might not be the host compiler!

Reply:
Also AppleClang take their branch cuts at a different time from main line Clang. There are feature mismatches